### ISSUE TITLE
no-else-return incorrectly marks else if statements as failing.

### ISSUE THREAD

**Issue by briandipalma** [eslint/eslint#675]
I found what appears to be a small issue with the `no-else-return` rule.
It seems to error even when you use an `else if` statement such as the code below.

``` js
    if (stats.isFile() && /.*\.js$/.test(path) === false) {
        return true;
    } else if (stats.isDirectory() && shouldDirectoryBeIgnored(path)) {
        return true;
    }
```

Produces

`50:8  error  Unexpected 'else' after 'return'​    no-else-return`

Removing the `else` would not be correct.
This is a trivial fix, I've found the issue and added a test to cover this case.
PR incoming soon unless I'm making a silly mistake.

**Comment by wilsonsilva** [comment]
This is still an issue.
